security/src/main/java/bisq/security/pow/ProofOfWorkService.java

```diff
@@ -35,7 +37,7 @@ public ProofOfWorkService() {
     public CompletableFuture<ProofOfWork> mintAsync(byte[] payload,
                                                     byte[] challenge,
                                                     double difficulty) {
-        return CompletableFuture.supplyAsync(() -> mint(payload, challenge, difficulty));
+        return CompletableFuture.supplyAsync(() -> mint(payload, challenge, difficulty), commonForkJoinPool());
```

Comment:
I guess here we should use a dedicated thread, but can be discussed in a new issue/PR.